diff --git a/build.mill b/build.mill
--- a/build.mill
+++ b/build.mill
@@ -18,3 +18,3 @@
 object v {
-  val scala      = "3.7.4"
+  val scala      = "3.8.4"
   val mainargs   = mvn"com.lihaoyi::mainargs:0.7.6"
@@ -30,5 +30,5 @@
 trait ZaoziScalaModule extends ScalaModule {
   override def scalacOptions: T[Seq[String]] = Task(
-    super.scalacOptions() ++ Seq("-java-output-version", "25")
+    super.scalacOptions() ++ Seq("-java-output-version", "25", "-Xsemanticdb", "-sourceroot", mill.api.BuildCtx.workspaceRoot.toString)
   )
 }
